Lilypad-Tech/lilypad
# feat: preflight check for bacalhau version

### Summary

Introduces a pre-flight check for resource-provider to check that bacalhau is running and a currently supported version (`1.3.2`).

### Task/Issue reference

Closes: https://github.com/Lilypad-Tech/lilypad/issues/280

### Test plan

- Run resource provider with bacalhau not running - should fial
- Run resource provider with bacalhau running but not v1.3.2 (e.g. 1.4.0) - should fail
- Run resource provider with bacalhau running at 1.3.2 - should run normally.
